### ISSUE TITLE
Suggested improvement for C++ day 1

### ISSUE THREAD

**Issue by tholok97** [Arxcis/adventofcode2018#3]
*I'm not sure what the etiquette on suggested improvements is in the repo, please let me know any opinions around that. Could be appropriate to discuss in the README as well.*

In `day01-chronal-calibration/main.cpp` a std::map is used to store seen frequencies. Wouldn't a std::set be more idiomatic, since the value stored for each element (1) is never used? The change would mostly be for readability. I'm not sure about the performance differences, but a quick google says they basically have the same lookup and insert times.

The Bash and Golang solutions use maps as well, but they don't have built-in sets to take advantage of.

**Comment by Arxcis** [comment]
Good point. In the Golang solution I really wanted a `Set` to solve the problem, but like you said there are no standard Set in golang. Since c++ has `#include <set>` it would be perfect to use it there.
